tests/README.md

```diff
@@ -0,0 +1,427 @@
+   if you modify your test to be able to run quicker and/or with less messages
+   if the `test_quick` variable is true.
+ * There's plenty of helper wrappers in test.c for common librdkafka functions
+   that makes tests easier to write by not having to deal with errors, etc.
+ * Fail fast, use `TEST_ASSERT()` et.al., the sooner an error is detected
+   the better since it makes troubleshooting easier.
+ * Use `TEST_SAY()` et.al. to inform the developer what your test is doing,
+   making it easier to troubleshoot upon failure. But try to keep output
+   down to reasonable levels. There is a `TEST_LEVEL` environment variable
+   that can be used with `TEST_SAYL()` to only emit certain printouts
+   if the test level is increased. The default test level is 2.
+
+
+
+## A guide to testing, verifying, and troubleshooting, librdkafka
+
+
+### Creating a development build
+
+The [dev-conf.sh](../dev-conf.sh) script configures and builds librdkafka and
+the test suite for development use, enabling extra runtime
+checks (`ENABLE_DEVEL`, `rd_dassert()`, etc), disabling optimization
+(to get accurate stack traces and line numbers), etc.
+It also makes it easy to enable ASAN or TSAN (see below) by
+
+    # Reconfigure librdkafka for development use and rebuild.
+    $ ./dev-conf.sh
+
+**NOTE**: Performance tests and benchmarks should not use a development build.
+
+
+### Controlling the test framework
+
+A test run may be dynamically set up using a number of environment variables.
+These environment variables work for all different ways of invocing the tests,
+be it `make`, `run-test.sh`, `until-fail.sh`, etc.
+
+ * `TESTS=0nnn` - only run a single test identified by its full number, e.g.
+                  `TESTS=0102 make`. (Yes, the var should have been called TEST)
+ * `TEST_DEBUG=...` - this will automatically set the `debug` config property
+                      of all instantiated clients to the value.
+                      E.g.. `TEST_DEBUG=broker,protocol TESTS=0001 make`
+ * `TEST_LEVEL=n` - controls the `TEST_SAY()` output level, a higher number
+                      yields more test output. Default level is 2.
+
+
+Let's say that you run the full test suite and get a failure in test 0061,
+which is a consumer test. You want to quickly reproduce the issue
+and figure out what is wrong, so limit the tests to just 0061, and provide
+the relevant debug options (which is typically `cgrp,fetch` for consumers):
+
+    $ TESTS=0061 TEST_DEBUG=cgrp,fetch make
+
+If the test did not fail you've found an intermittent issue, this is where
+[until-fail.sh](until-fail.sh) comes in to play, so run the test until it fails:
+
+    # bare means to run the test without valgrind
+    $ TESTS=0061 TEST_DEBUG=cgrp,fetch ./until-fail.sh bare
+
+
+### How to run tests
+
+The standard way to run the test suite is firing up a trivup cluster
+in an interactive shell:
+
+    $ ./interactive_broker_version.py 2.3.0   # Broker version
```

Comment:
maybe document `trunk` explicitly here.

Reply:
I tried to ignore trunk runs since they are more complex and error prone.
Do you think we need them documented here?

Reply:
i mentioned it because it would have been helpful for me with #2518 and we'll more than likely need to test against trunk in the future as we move to working in parallel with java.

Reply:
Makes sense, I'd like to add that in another PR later though since I think there are some necessary improvements to source paths, etc.

Reply:
sounds good